clients/src/main/java/org/apache/kafka/common/security/JaasUtils.java

```diff
@@ -17,107 +17,22 @@
-                return jaasConfig;
-            }
-        } else
-            return defaultJaasConfig(loginType);
-    }
-
-    private static Configuration defaultJaasConfig(LoginType loginType) {
-        String jaasConfigFile = System.getProperty(JaasUtils.JAVA_LOGIN_CONFIG_PARAM);
-        if (jaasConfigFile == null) {
-            LOG.debug("System property '" + JaasUtils.JAVA_LOGIN_CONFIG_PARAM + "' and Kafka SASL property '" +
-                      SaslConfigs.SASL_JAAS_CONFIG + "' are not set, using default JAAS configuration.");
-        }
-
-        Configuration jaasConfig = Configuration.getConfiguration();
-
-        String loginContextName = loginType.contextName();
-        AppConfigurationEntry[] configEntries = jaasConfig.getAppConfigurationEntry(loginContextName);
-        if (configEntries == null) {
-            String errorMessage;
-            errorMessage = "Could not find a '" + loginContextName + "' entry in the JAAS configuration. System property '" +
-                    JaasUtils.JAVA_LOGIN_CONFIG_PARAM + "' is " + (jaasConfigFile == null ? "not set" : jaasConfigFile);
-            throw new IllegalArgumentException(errorMessage);
-        }
-        return jaasConfig;
-    }
-
-    /**
-     * Returns the configuration option for <code>key</code> from the server login context
-     * of the default JAAS configuration. If login module name is specified, return option value
-     * only from that module.
-     */
-    public static String defaultServerJaasConfigOption(String key, String loginModuleName) throws IOException {
-        return jaasConfigOption(Configuration.getConfiguration(), LoginType.SERVER.contextName(), key, loginModuleName);
-    }
-
-    /**
-     * Returns the configuration option for <code>key</code> from the login context
-     * <code>loginContextName</code> of the specified JAAS configuration.
-     * If login module name is specified, return option value only from that module.
-     */
-    public static String jaasConfigOption(Configuration jaasConfig, String loginContextName, String key, String loginModuleName) throws IOException {
-        AppConfigurationEntry[] configurationEntries = jaasConfig.getAppConfigurationEntry(loginContextName);
-        if (configurationEntries == null) {
-            String errorMessage = "Could not find a '" + loginContextName + "' entry in this JAAS configuration.";
-            throw new IOException(errorMessage);
-        }
-
-        for (AppConfigurationEntry entry: configurationEntries) {
-            if (loginModuleName != null && !loginModuleName.equals(entry.getLoginModuleName()))
-                continue;
-            Object val = entry.getOptions().get(key);
-            if (val != null)
-                return (String) val;
-        }
-        return null;
-    }
-
     public static String defaultKerberosRealm()
```

Comment:
This should probably be moved somewhere else since it doesn't come from jaas config. And then the only method left in this class would be `isZkSecurityEnabled` which could probably move to `core`.